packages/components/src/components/hds/form/super-select/single/base.ts

```diff
@@ -0,0 +1,144 @@
+
+export interface HdsFormSuperSelectSingleBaseSignature {
+  Args: PowerSelectSignature['Args'] & {
+    showAfterOptions?: boolean;
+    afterOptionsContent?: string;
+    resultCountMessage?: string;
+    dropdownMaxWidth?: string;
+    matchTriggerWidth?: boolean;
+    isInvalid?: boolean;
+  };
+  Blocks: PowerSelectSignature['Blocks'];
+  Element: PowerSelectSignature['Element'];
+}
+
+export default class HdsFormSuperSelectSingleBaseComponent extends Component<HdsFormSuperSelectSingleBaseSignature> {
+  @tracked powerSelectAPI?: PowerSelect;
+
+  get horizontalPosition(): HdsFormSuperSelectHorizontalPositions {
+    const { horizontalPosition = DEFAULT_HORIZONTAL_POSITION } = this.args;
+    return horizontalPosition as HdsFormSuperSelectHorizontalPositions;
+  }
+
+  get resultCountMessage(): string {
+    return `${this.powerSelectAPI?.resultsCount || 0} total`;
+  }
+
+  /**
+   * This action sets the powerSelectAPI property and optionally calls a registerAPI function.
+   *
+   * @param {Object} powerSelectAPI - The API object for the PowerSelect component.
+   *
+   * If a `registerAPI` function is passed in through the component's arguments,
+   * this function will be called with the `powerSelectAPI` as its argument.
+   * This allows parent components or controllers to have access to the PowerSelect API.
+   *
+   * The `powerSelectAPI` is also stored on the component instance and used in `clearSelected`
+   */
+  @action
+  setPowerSelectAPI(powerSelectAPI: PowerSelect): void {
+    if (typeof this.args.registerAPI === 'function') {
+      this.args.registerAPI(powerSelectAPI);
+    }
+    this.powerSelectAPI = powerSelectAPI;
+  }
+
+  @action calculatePosition(
+    trigger: Element,
+    content: HTMLElement
+  ): CalculatePositionResult {
+    // use `hds-anchored-position` to calculate and set position
+    // @ts-expect-error: known issue with type of invocation
+    anchoredPositionModifier(content, [trigger], {
+      placement: HORIZONTAL_POSITION_MAPPING[this.horizontalPosition],
+      offsetOptions: 4,
+      enableCollisionDetection: true,
+    });
+    // prevent PowerSelect from setting position
+    return {
+      horizontalPosition: 'auto',
+      verticalPosition: 'auto',
+      style: {},
+    };
+  }
+
+  /**
+   * Determine if `@afterOptionsComponent` gets displayed
+   * @param showAfterOptions
+   * @type {boolean}
+   * @default false
+   */
+  get showAfterOptions(): boolean | string {
+    return this.args.showAfterOptions ?? this.args.afterOptionsContent ?? false;
+  }
+
+  /**
+   * Get the search placeholder text
+   * @param searchPlaceholder
+   * @type {string}
+   * @default 'Search'
+   */
+  get searchPlaceholder(): string {
+    return this.args.searchPlaceholder ?? 'Search';
+  }
+
+  /**
+   * Get the maxWidth to apply to the dropdown
+   * @param dropdownMaxWidth
+   * @type {string}
+   * @default 'none'
+   */
+  get dropdownMaxWidthStyle() {
```

Comment:
Nit: missing this return type

Reply:
Good catch! Added in https://github.com/hashicorp/design-system/pull/2356/commits/08c99afae38f74beb19294c618baf00e1eb2ee5e